test/WebJobs.Script.Tests/ScriptHostTests.cs

```diff
@@ -891,17 +893,28 @@ public void Initialize_LogsWarningForExplicitlySetHostId()
                 RootScriptPath = rootPath
             };
 
-            // This id will be over written
-            // TODO: DI (FACAVAL) Review
-            //config.HostConfig.HostId = ID;
-            var environment = new Mock<IScriptJobHostEnvironment>();
-            var eventManager = new Mock<IScriptEventManager>();
+            var host = new HostBuilder()
+                .ConfigureDefaultTestWebScriptHost(_ => { },
+                options =>
+                {
+                    options.ScriptPath = rootPath;
+                    options.LogPath = Path.GetTempPath();
+                }, false,
+                rootServices =>
+                {
+                    rootServices.AddSingleton<ILoggerFactory>(loggerFactory);
+                })
+                .Build();
+
+            await host.StartAsync();
+
+            var idProvider = host.Services.GetService<IHostIdProvider>();
+            string hostId = await idProvider.GetHostIdAsync(CancellationToken.None);
```

Comment:
Important test we were missing. The behavior was correct, but this validates that IDs defined in host.json are honored by the ID provider and used throughout the application.